File: app/src/main/java/android/mahendra/attendancemanager/viewmodels/PeriodViewModel.java
package android.mahendra.attendancemanager.viewmodels;

import android.mahendra.attendancemanager.models.Period;

import androidx.databinding.BaseObservable;

public class PeriodViewModel extends BaseObservable {
    private Period mPeriod;

    public void setPeriod(Period period) {
        mPeriod = period;
    }

    public String getTitle() {
        if (mPeriod == null) {
            return "-";
        }
        return mPeriod.getSubjectTitle();
    }

    public String getPeriodNumber() {
        if (mPeriod == null) {
            return "-";
        }
        return String.valueOf(mPeriod.getPeriodNumber());
    }
}

File: app/src/main/java/android/mahendra/attendancemanager/viewmodels/PeriodViewModel.kt
package android.mahendra.attendancemanager.viewmodels

import android.mahendra.attendancemanager.models.Period

import androidx.databinding.BaseObservable

class PeriodViewModel(private var period: Period? = null) : BaseObservable() {

    fun setPeriod(period: Period?) {
        this.period = period
    }

    val title: String?
        get() = period!!.subjectTitle

    val periodNumber: String
        get() = period!!.periodNumber.toString()
}
